### PR TITLE
Add ADR to control applicant access using Delius caseload

### PR DESCRIPTION
**Pull request by edavey**

### CODE REVIEW

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (brownbearsoftware) [inline doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:157]: I don't think this is as much of an issue anymore as we will know which of the user's teams manages the Offender from the response from `https://approved-premises-and-delius/{crn}/managingTeams/{delius-staffcode}`.

The edge case is reduced to: the user is a member of more than one team where more than one of those teams has the Offender in its caseload.  Ideally I think we would ask the user to select the relevant team but given time constraints perhaps just storing all of the teams is the way to go here?

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (pezholio) [inline doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:27]: Should we add something about continuing applications that they have made? Or do we want them to be able to continue applications their teammates have made? Though this could get messy.

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (pezholio) [inline doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:42]: Just flagging: This isn't functionality we support yet

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (pezholio) [inline doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:50]: Assessors _will_ be able to ask an applicant for further info too

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (pezholio) [inline doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:21]: Should this be part of the decision?

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (pezholio) [inline doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:111]: I feel like a more RESTful approach would be something like:

https://approved-premises-and-delius/teams/search

With a JSON payload eg:

```json
{ "crn": "{crn}", "staffCode": "{delius-staffcode}" }
```

Though I guess the implementation is up to the integrations team

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (pezholio) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:157]: I think we'll need to understand if there is ever a case where an offender is in more than one caseload first? It seems to me like it'd be a 1:1 relationship right?

Comment (brownbearsoftware) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:157]: I think this depends on the answer to point 2.  i.e. an Offender can appear in the caseload for multiple teams at more granular levels so it depends which level of granulairty the integrations team endpoint operates at.  If I recall correctly this was examplified as part of their probation terms might be the responsibilit of a different team to the one that bares overall responsibility for them

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (marcus-bcl) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:111]: Another option would be to use query parameters as filters:
```
GET /teams?managesCrn={crn}&hasStaffMember={delius-staffcode}
{
    "teams": ["N01ABC"]
}
```

But we're happy to use whichever implementation - does your team have a preference?

Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:111]: Thanks @marcus-bcl 

@brownbearsoftware Would our OpenAPI code generator have trouble with any of these approaches? 

If not then probably it makes sense for you to decide on a design which is consistent within your mono-repo?

Comment (brownbearsoftware) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:111]: We'd be consuming this endpoint so it wouldn't need to touch OpenAPI Generate so any approach will work for us 👍

Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:111]: good point! I forget which side of the fence I'm on!

Comment (marcus-bcl) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:111]: We ended up going with:
```
GET /teams/managingCase/{crn}?staffCode={staffCode}
{
  "teamCodes": ["N01ABC"]
}
```

Hope that works for you.  The endpoint is up for review now, so should be available to test soon.

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:157]: Yes I believe that more than one team can have a person in their caseload so the endpoint may return more than one team. And it sounds like storing a list of team codes against each application (e.g. `application.managing_team_codes`) would be a good solution.

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:27]: I think we've generally agreed that:

- the creator of a application can edit it until it's been submitted and then it's read-only -- apart from the ability to add clarification notes (the form of these post-submission exchanges between assessor and applicant is not finalised)
- collaborating on an application is out of scope, i.e. if I am struck by lightning halfway through creating an application then a colleague will need to create a new application from scratch

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md:21]: hmm, I thought it was helpful to include in the "Context" section as background, but on reflection I think I agree that "less is more" and that it's context enough to say:

- we've got some internally managed rolew
- we've got one special default role whose capabilities are defined by team caseloads managed within Delius

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md]: removed all this details from the ADR

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md]: Removed this level of detail

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md]: this looks good, thanks

File: doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md
Comment (edavey) [reply doc/architecture/decisions/0011-control-applicant-access-using-delius-team-caseload.md]: I've added this to the draft

**Comment by edavey** [comment]
@pezholio @brownbearsoftware 
I've updated this draft to:

- remove some background detail of dubious relevance
- include details of the actual endpoint implementation
- incorporate the latest thinking on handling multiple team codes
- a note about inter-PDU applications

**Review by pezholio — DISMISSED** [review]

**Comment by edavey** [comment]
@pezholio can I get a re-approval please as this went stale before I could merge!

**Pull request merged**
